crates/apub/src/objects/community.rs

```diff
@@ -213,7 +220,21 @@ impl Object for ApubCommunity {
         featured.dereference(&community_, &context_).await.ok();
       }
       if let Some(moderators) = group.attributed_to {
-        moderators.dereference(&community_, &context_).await.ok();
+        if let AttributedTo::Lemmy(l) = moderators {
+          l.moderators()
+            .dereference(&community_, &context_)
+            .await
+            .ok();
+        } else if let AttributedTo::Peertube(p) = moderators {
+          let new_mods = p
+            .iter()
+            .filter(|p| p.kind == PersonOrGroupType::Person)
+            .map(|p| ObjectId::<ApubPerson>::from(p.id.clone().into_inner()))
```

Comment:
There's also a [`filter_map`](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.filter_map) function that can combine these 2 steps:

```suggestion
            .filter_map(|p| (p.kind == PersonOrGroupType::Person)
                .then(|| ObjectId::<ApubPerson>::from(p.id.clone().into_inner()))
            )
```

If the type that `p.id` implements `Copy`, you can also omit the call to `clone`.

Reply:
Clippy changes this back to a `filter` and `map`. https://rust-lang.github.io/rust-clippy/master/index.html#filter_map_bool_then